pkg/apiserver/rest/webservice/user.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webservice
+
+import (
+	"context"
+
+	restfulspec "github.com/emicklei/go-restful-openapi/v2"
+	"github.com/emicklei/go-restful/v3"
+
+	"github.com/oam-dev/kubevela/pkg/apiserver/model"
+	apis "github.com/oam-dev/kubevela/pkg/apiserver/rest/apis/v1"
+	"github.com/oam-dev/kubevela/pkg/apiserver/rest/usecase"
+	"github.com/oam-dev/kubevela/pkg/apiserver/rest/utils"
+	"github.com/oam-dev/kubevela/pkg/apiserver/rest/utils/bcode"
+)
+
+type userWebService struct {
+	userUsecase usecase.UserUsecase
+}
+
+// NewUserWebService is the webservice of user
+func NewUserWebService(userUsecase usecase.UserUsecase) WebService {
+	return &userWebService{
+		userUsecase: userUsecase,
+	}
+}
+
+func (c *userWebService) GetWebService() *restful.WebService {
+	ws := new(restful.WebService)
+	ws.Path(versionPrefix+"/users").
+		Consumes(restful.MIME_XML, restful.MIME_JSON).
+		Produces(restful.MIME_JSON, restful.MIME_XML).
+		Doc("api for user manage")
+
+	tags := []string{"users"}
+
+	ws.Route(ws.GET("/").To(c.listUser).
+		Doc("list users").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Filter(c.userCheckFilter).
+		Param(ws.QueryParameter("page", "query the page number").DataType("integer")).
+		Param(ws.QueryParameter("pageSize", "query the page size number").DataType("integer")).
+		Param(ws.QueryParameter("name", "fuzzy search based on name").DataType("string")).
+		Param(ws.QueryParameter("email", "fuzzy search based on email").DataType("string")).
+		Param(ws.QueryParameter("alias", "fuzzy search based on alias").DataType("string")).
+		Returns(200, "OK", apis.ListUserResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(apis.ListUserResponse{}))
+
+	ws.Route(ws.POST("/").To(c.createUser).
+		Doc("create a user").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(apis.CreateUserRequest{}).
+		Returns(200, "OK", apis.DetailUserResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(apis.DetailUserResponse{}))
```

Comment:
return the DetailUserResponse but the projects do not set values, inaccurate for generated API documentation.